main.go

```diff
@@ -158,47 +136,13 @@ func mainWithExitCode() exitCode {
 		return failure("Failed to probe tracepoint: %v", err)
 	}
 
-	var presentCores uint16
-	presentCores, err = hostmeta.PresentCPUCores()
+	presentCores, err := numcpus.GetOnline()
```

Comment:
It seems like there's a semantic change here as present CPU cores != online CPU cores and since we're sizing by worst case, we should probably keep using present cores?

Reply:
Resolved with https://github.com/elastic/otel-profiling-agent/pull/90/commits/654c6366477b5d88ccea4e3ca40dbab3f8c906fe.